numpy/core/records.py

```diff
@@ -169,8 +169,13 @@ def _parseFormats(self, formats, aligned=0):
         if isinstance(formats, list):
             if len(formats) < 2:
                 formats.append('')
-            formats = ','.join(formats)
-        dtype = sb.dtype(formats, aligned)
+            dtype = sb.dtype(
+                {'formats': formats,
```

Comment:
The route via `dict` is quite a bit slower than via a list, so maybe:
```
dtype = sb.dtype([('f{}'.format(i), format_) for i, format_ in enumerate(formats)])
```

Timings:
```
formats = ['f8', 'i8']
%timeit np.dtype({'formats': formats, 'names': ['f{}'.format(i) for i in range(len(formats))]})
# 100000 loops, best of 3: 3.44 µs per loop
%timeit np.dtype([('f{}'.format(i), format_) for i, format_ in enumerate(formats)])
# 100000 loops, best of 3: 2.24 µs per loop
```

(And @eric-wieser is correct, no need to append '' anymore)